src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java

```diff
@@ -0,0 +1,78 @@
+package name.webdizz.sonar.grammar.issue.tracking;
+
+import com.google.common.base.Preconditions;
+import name.webdizz.sonar.grammar.GrammarPlugin;
+import name.webdizz.sonar.grammar.PluginParameter;
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.ServerExtension;
+import org.sonar.api.config.Settings;
+import org.sonar.api.issue.action.Function;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.apache.commons.lang.StringUtils.stripToEmpty;
+
+public class LinkFunction implements Function, ServerExtension {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinkFunction.class);
+
+    @Override
+    public void execute(Context context) {
+        String word = getMistakeWord(context.issue().message(), PluginParameter.ERROR_DESCRIPTION);
+        LOGGER.info("try to write:" + word);
+        Settings settings = context.projectSettings();
+        String dict = stripToEmpty(settings.getString(PluginParameter.MANUAL_DICT));
+        saveProperties(dict + "," + word, PluginParameter.MANUAL_DICT, context);
+    }
+
+    private String getMistakeWord(String message, String errDescription) {
+        String res = message.replaceFirst(errDescription, "");
+        return res.substring(0, res.length() - 1);
+    }
+
+    private void saveProperties(String word, String dictName, Context context) {
```

Comment:
do not agree with this implementation - why do we need to use remote api when we're installed within Sonar? there should be an API to write properties without making a connection over http